### DISCUSSION TITLE
How to run loader only once?

### DISCUSSION THREAD

**Discussion by juniorforlife** [remix-run/remix#6214] [Q&A] [answered]
I have an API to load videos which would be really slow if I query all videos at once. 
To speed things up, I call this API twice in the loader, one with a limit of only 10 videos and one with all videos. Then I use `defer` for the heavy API call to not slow down the page loading time
```
export async function loader(){
  const quickVideos = await getVideos({ limit: 10 })
  const fullVideos = getVideos({limit: undefined}) // fetch all
  return defer({ quickVideos, fullVideos });
}
  
```
The problem is I want the loader to run only once for the initial page load. Whenever an action happens, I dont want to re-run the 2 API again because it's not necessary. I can use optimistic UI to update instantly rather than waiting for the whole loader to run and re-render the page. How do I achieve this?

**Comment by sergiodxa** [comment, marked as answer]
You can use the [`shouldRevalidate` export](https://remix.run/docs/en/1.15.0/route/should-revalidate) to tell Remix to never revalidate it, or only do it under certain specific conditions (like a special search param changing).

Also I recommend you to try to cache those requests so you can let the loader run many times but still fetch once, a server-side cache will also let you share the data between users which means if you receive 100 users only one will have to wait for the request, compared to every one doing it.

**Reply by juniorforlife** [reply]
You're super useful as always. Thank you!
